src/core/tests/Test_RebootManager.py

```diff
@@ -74,6 +74,22 @@ def test_reboot_always_time_available(self):
         self.assertEqual(reboot_manager.start_reboot_if_required_and_time_available(20), True)
         runtime.stop()
 
+    def test_reboot_always_runs_only_once_if_no_reboot_is_required(self):
+        reboot_setting_in_api = 'Always'
+        argument_composer = ArgumentComposer()
+        argument_composer.reboot_setting = reboot_setting_in_api
+        runtime = RuntimeCompositor(argument_composer.get_composed_arguments(), True, Constants.YUM)
+        reboot_manager = runtime.reboot_manager
+
+        # mock completing the reboot once, with no reboot required
+        runtime.status_handler.set_installation_reboot_status(Constants.RebootStatus.REQUIRED)
+        runtime.status_handler.set_installation_reboot_status(Constants.RebootStatus.STARTED)
+        runtime.status_handler.set_installation_reboot_status(Constants.RebootStatus.COMPLETED)
+
```

Comment:
Is there a way to ensure reboot_pending is True, just so that this test always hits that scenario?

Reply:
The test has been modified to go through the first reboot scenario and then no reboot with the same setting